### PR TITLE
feat(engine): save form to the state

### LINKED ISSUE #14133: Save Form to the state

### LINKED ISSUE #14137: Implement query to retrieve the latest form by formId from the state

### PR DESCRIPTION
**Pull request by berkaycanbc**
This PR includes:

* Form DB classes
* Form column family definitions
* A method to store the Form
* A method to retrieve the latest form by form id
* A method to retrieve a Form by key
* Tests for implemented repository methods

## Related issues

closes #14133 
closes #14137 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by nicpuppa — APPROVED** [review]
LGTM 🚀 

There is a test failure, I'll give a look 👀

File: protocol/src/main/java/io/camunda/zeebe/protocol/record/value/deployment/Form.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.protocol.record.value.deployment;
+
+import io.camunda.zeebe.protocol.record.ImmutableProtocol;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@ImmutableProtocol(builder = ImmutableProcess.Builder.class)
```
Comment (nicpuppa) [inline protocol/src/main/java/io/camunda/zeebe/protocol/record/value/deployment/Form.java:22]: ❌ I give a look at the failed test and the error is pretty clear

```suggestion
@ImmutableProtocol(builder = ImmutableForm.Builder.class)
```

**Review by nicpuppa — CHANGES_REQUESTED** [review]

**Review by nicpuppa — APPROVED** [review]
LGTM 💪

**Comment by berkaycanbc** [comment]
bors merge

**Pull request merged**